### PR TITLE
Make telemetry opt-out

### PR DESCRIPTION
**Pull request by 6za**
Allow telemetry to be disabled by user choice. 

Signed-off-by: 6za <53096417+6za@users.noreply.github.com>

### CODE REVIEW

**Comment by converge** [comment]
it would be better to use the dry run flag to decide to send/or not telemetry data

`send telemetry data only when dry-run flag is off`

then we dont need extra flags/code

**Comment by converge** [comment]
> * Telemetry and Dry-run are not related, it just don't change anything outside, it is just a general run test without real action.

why do we want to have a real run without telemetry?

File: cmd/init.go [hunk 1]
```diff
@@ -221,4 +229,5 @@ func init() {
 
 	initCmd.Flags().String("cluster-name", "kubefirst", "the cluster name, used to identify resources on cloud provider")
 	initCmd.Flags().String("version-gitops", "main", "version/branch used on git clone")
+	initCmd.Flags().Bool("disable-telemetry", false, "Installer will not send telemetry about this installation")
```
Comment (converge) [inline cmd/init.go:232]: what about name it `telemetry`, and have it enabled by default?

user calls `--telemetry=false` when he decides to opt-out telemetry

File: cmd/root.go [hunk 2]
```diff
@@ -32,5 +32,6 @@ func init() {
 	cobra.OnInitialize()
 
 	// Cobra also supports local flags, which will only run, when this action is called directly.
-	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
```
Comment (converge) [inline cmd/root.go:35]: left over?

File: internal/telemetry/telemetry.go [hunk 3]
```diff
@@ -9,33 +9,37 @@ import (
 )
 
 // SendTelemetry post telemetry data
-func SendTelemetry(domain, metricName string) {
-	log.Println("SendTelemetry (working...)")
+func SendTelemetry(disableTelemetry bool, domain string, metricName string) {
+	if !disableTelemetry {
```
Comment (converge) [inline internal/telemetry/telemetry.go:13]: what about:
```go
if !sendTelemetry {
  return
}

... send telemetry code ...
```

**Comment by 6za** [comment]
Updated sample:

```
developer@6c3b5f506205:~/app$ go run . cluster create --dry-run=true  --use-telemetry=true
Init actions: 4 expected tasks ...

LOG: 2022/07/20 20:16:14.601942 /home/developer/app/cmd/init.go:229: init started
Logging at: /home/developer/app/logs/log_1658348174.log
LOG: 2022/07/20 20:16:14.602344 /home/developer/app/pkg/helpers.go:132: Using config file: /home/developer/.kubefirst
-----------


  ----------------------------------------------------------------------
  Info summary:
  ----------------------------------------------------------------------

  Operational System: linux
  Architecture: amd64
  Go Lang version: vgo1.17.11
  Kubefirst config file: /home/developer/.kubefirst
  Kubefirst config folder: /home/developer/.k1
  Kubectl path: /home/developer/.k1/tools/kubectl
  Terraform path: /home/developer/.k1/tools/terraform
  Kubeconfig path: /home/developer/.k1/gitops/terraform/base/kubeconfig



Process Parameters            ... done! [1 in 1.004s]
- Creating K8S Cluster
- Created Softserve
Prepare Temporary Repo        ... 50.0% [##############..............] [2 in 2.69109s; ~ETA: 3s]
^Csignal: interrupt
developer@6c3b5f506205:~/app$ go run . cluster create --dry-run=true  --use-telemetry=false
Init actions: 4 expected tasks ...

LOG: 2022/07/20 20:16:25.293513 /home/developer/app/cmd/init.go:229: init started
Logging at: /home/developer/app/logs/log_1658348185.log
LOG: 2022/07/20 20:16:25.294123 /home/developer/app/pkg/helpers.go:132: Using config file: /home/developer/.kubefirst
-----------


  ----------------------------------------------------------------------
  Info summary:
  ----------------------------------------------------------------------

  Operational System: linux
  Architecture: amd64
  Go Lang version: vgo1.17.11
  Kubefirst config file: /home/developer/.kubefirst
  Kubefirst config folder: /home/developer/.k1
  Kubectl path: /home/developer/.k1/tools/kubectl
  Terraform path: /home/developer/.k1/tools/terraform
  Kubeconfig path: /home/developer/.k1/gitops/terraform/base/kubeconfig



Process Parameters            ... done! [1 in 1.003s]
- Telemetry Disabled
- Creating K8S Cluster
- Created Softserve
- Waiting Softserve
Prepare Temporary Repo        ... 75.0% [#####################.......] [3 in 3.190274s]
```

**Review by johndietz — APPROVED** [review]

**Pull request merged**